File: leetcode/88/P88.py
# 88. Merge Sorted Array
class Solution(object):
	def merge(self, nums1, m, nums2, n):
		i = j = 0
		nums3 = []
		while i < m and j < n:
			if nums1[i] < nums2[j]:
				nums3.append(nums1[i])
				i += 1
			else:
				nums3.append(nums2[j])
				j += 1
		nums3.extend(nums1[i:m])
		nums3.extend(nums2[j:n])
		del nums1[:]
		nums1.extend(nums3)

l = [1]
Solution().merge(l, 1, [], 0)
print l

File: leetcode/88/src/P88.java
/**
 * Created by yfy on 2/4/16.
 * 88. Merge Sorted Array
 */

class Solution {
	public void merge(int[] nums1, int m, int[] nums2, int n) {
		int i, j, k;
		int[] nums3 = new int[m + n];
		i = j = k = 0;
		while (i < m && j < n) {
			if (nums1[i] < nums2[j]) {
				nums3[k++] = nums1[i];
				i++;
			} else {
				nums3[k++] = nums2[j];
				j++;
			}
		}
		System.arraycopy(nums1, i, nums3, k, m - i);
		System.arraycopy(nums2, j, nums3, k + m - i, n - j);
		System.arraycopy(nums3, 0, nums1, 0, m + n);
	}
}

public class P88 {

}
